### PR TITLE
[Merged by Bors] - feat: define instances of `Finite`, `Fintype`, `DecidableEq` for `MulEquiv`

### PR DESCRIPTION
**Pull request by wupr**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Fintype/Perm.lean [hunk 1]
```diff
@@ -140,16 +140,26 @@ theorem card_perms_of_finset : ∀ s : Finset α, #(permsOfFinset s) = (#s)! :=
 def fintypePerm [Fintype α] : Fintype (Perm α) :=
   ⟨permsOfFinset (@Finset.univ α _), by simp [mem_perms_of_finset_iff]⟩
 
-instance equivFintype [Fintype α] [Fintype β] : Fintype (α ≃ β) :=
+instance Equiv.fintype [Fintype α] [Fintype β] : Fintype (α ≃ β) :=
```
Comment (kim-em) [inline Mathlib/Data/Fintype/Perm.lean:143]: I think the convention here is `Equiv.instFintype`.

**Comment by kim-em** [comment]
bors d+

File: Mathlib/Data/Fintype/Perm.lean (hunk 1 above)
Comment (wupr) [reply Mathlib/Data/Fintype/Perm.lean:143]: I see. Does this mean my `MulEquiv.fintype` should also be `MulEquiv.instFintype`?

**Pull request closed without merging**